From 37b12cb2b4dcc74805aa61bfb4566dba1e671a72 Mon Sep 17 00:00:00 2001
From: Dan Gastler <dgastler@gmail.com>
Date: Sun, 29 Nov 2020 13:42:21 -0500
Subject: [PATCH] fixes

---
 mmc_cmds.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mmc_cmds.c b/mmc_cmds.c
index fb37189..95df97c 100644
--- a/mmc_cmds.c
+++ b/mmc_cmds.c
@@ -1915,7 +1915,7 @@ static int do_rpmb_op(int fd,
 	u_int16_t rpmb_type;
 	struct mmc_ioc_multi_cmd *mioc;
 	struct mmc_ioc_cmd *ioc;
-	struct rpmb_frame frame_status = {0};
+	struct rpmb_frame frame_status = {{0}};
 
 	if (!frame_in || !frame_out || !out_cnt)
 		return -EINVAL;
-- 
1.8.3.1

