diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
index 434b9c8..b3bd1ee 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_config.h
@@ -86,9 +86,9 @@ extern "C" {
  *       all data exchanged.
  */
 #define FSBL_PRINT_VAL              (1U)
-#define FSBL_DEBUG_VAL              (0U)
-#define FSBL_DEBUG_INFO_VAL         (0U)
-#define FSBL_DEBUG_DETAILED_VAL     (0U)
+#define FSBL_DEBUG_VAL              (1U)
+#define FSBL_DEBUG_INFO_VAL         (1U)
+#define FSBL_DEBUG_DETAILED_VAL     (1U)
 
 /**
  * FSBL Debug options
diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_sd.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_sd.c
index accebea..de63375 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_sd.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_sd.c
@@ -89,6 +89,43 @@ u32 XFsbl_SdInit(u32 DeviceFlags)
 	char *boot_file = buffer;
 	u32 MultiBootOffset;
 	u32 DrvNum;
+	u32 ireg;
+	u32 jreg;
+        /*                                                                                                                                                                
+          test debugging of SDCARD                                                                                                                                        
+	*/
+        XFsbl_Printf(DEBUG_INFO,"    PS_VERSION              (0x%08X)        = 0x%08X\n\r",0xFFCA0044, XFsbl_In32(    0xFFCA0044)             );
+        XFsbl_Printf(DEBUG_INFO,"    BOOT_MODE_USER          (0x%08X)        = 0x%08X\n\r",0xFF5E0200, XFsbl_In32(    0xFF5E0200)             );
+        XFsbl_Printf(DEBUG_INFO,"    BOOT_MODE_POR           (0x%08X)        = 0x%08X\n\r",0xFF5E0204, XFsbl_In32(    0xFF5E0204)             );
+        XFsbl_Printf(DEBUG_INFO,"    RESET_REASON            (0x%08X)        = 0x%08X\n\r",0xFF5E0220, XFsbl_In32(    0xFF5E0220)             );
+        XFsbl_Printf(DEBUG_INFO,"    PMU_GLOBAL.PWR_STATE    (0x%08X)        = 0x%08X\n\r",0xFFD80100, XFsbl_In32(    0xFFD80100)             );
+        XFsbl_Printf(DEBUG_INFO,"    PWR_SUPPLY_STATUS       (0x%08X)        = 0x%08X\n\r",0xFFD8010C, XFsbl_In32(    0xFFD8010C)             );
+        XFsbl_Printf(DEBUG_INFO,"    CSU_BR_ERROR            (0x%08X)        = 0x%08X\n\r",0xFFD80528, XFsbl_In32(    0xFFD80528)             );
+        XFsbl_Printf(DEBUG_INFO,"    ERROR_STATUS_1          (0x%08X)        = 0x%08X\n\r",0xFFD80530, XFsbl_In32(    0xFFD80530)             );
+        XFsbl_Printf(DEBUG_INFO,"    ERROR_STATUS_2          (0x%08X)        = 0x%08X\n\r",0xFFD80540, XFsbl_In32(    0xFFD80540)             );
+        XFsbl_Printf(DEBUG_INFO,"    csu_status              (0x%08X)        = 0x%08X\n\r",0xFFCA0000, XFsbl_In32(    0xFFCA0000)             );
+        XFsbl_Printf(DEBUG_INFO,"    csu_ft_status           (0x%08X)        = 0x%08X\n\r",0xFFCA0018, XFsbl_In32(    0xFFCA0018)             );
+        XFsbl_Printf(DEBUG_INFO,"    CSU_ISR                 (0x%08X)        = 0x%08X\n\r",0xFFCA0020, XFsbl_In32(    0xFFCA0020)             );
+        XFsbl_Printf(DEBUG_INFO,"    pcap_status             (0x%08X)        = 0x%08X\n\r",0xFFCA3010, XFsbl_In32(    0xFFCA3010)             );
+        XFsbl_Printf(DEBUG_INFO,"    tamper_status           (0x%08X)        = 0x%08X\n\r",0xFFCA5000, XFsbl_In32(    0xFFCA5000)             );
+        XFsbl_Printf(DEBUG_INFO,"    jtag_chain_status       (0x%08X)        = 0x%08X\n\r",0xFFCA0034, XFsbl_In32(    0xFFCA0034)             );
+        XFsbl_Printf(DEBUG_INFO,"    jtag_sec                (0x%08X)        = 0x%08X\n\r",0xFFCA0038, XFsbl_In32(    0xFFCA0038)             );
+        XFsbl_Printf(DEBUG_INFO,"    sd0 status              (0x%08X)        = 0x%08X\n\r",0xFF160024, XFsbl_In32(    0xFF160024)             );
+        XFsbl_Printf(DEBUG_INFO,"    sd0 config              (0x%08X)        = 0x%02X\n\r",0xFF160028, XFsbl_In32(    0xFF160028) & 0xFF      );
+        XFsbl_Printf(DEBUG_INFO,"    sd0 power               (0x%08X)        = 0x%02X\n\r",0xFF160028,(XFsbl_In32(    0xFF160028) & 0xFF00)>>8);
+        XFsbl_Printf(DEBUG_INFO,"    sd1 status              (0x%08X)        = 0x%08X\n\r",0xFF170024, XFsbl_In32(    0xFF170024)             );
+        XFsbl_Printf(DEBUG_INFO,"    sd1 config              (0x%08X)        = 0x%02X\n\r",0xFF170028, XFsbl_In32(    0xFF170028) & 0xFF      );
+        XFsbl_Printf(DEBUG_INFO,"    sd1 power               (0x%08X)        = 0x%02X\n\r",0xFF170029,(XFsbl_In32(    0xFF170028) & 0xFF00)>>8);
+
+	for(ireg=0xFF170000;ireg < 0xFF1700FF;){
+	  XFsbl_Printf(DEBUG_INFO,"0x%08X: 0x",ireg);
+	  for(jreg = 0; jreg < 8; jreg++){
+	    XFsbl_Printf(DEBUG_INFO,"%08X ",XFsbl_In32(ireg));
+	    ireg+=4;
+	  }
+	  XFsbl_Printf(DEBUG_INFO,"\n\r");
+	}
+
 
 	DrvNum = XFsbl_GetDrvNumSD(DeviceFlags);
 
@@ -135,6 +172,17 @@ u32 XFsbl_SdInit(u32 DeviceFlags)
 	}
 
 	Status = XFSBL_SUCCESS;
+
+	for(ireg=0xFF170000;ireg < 0xFF1700FF;){
+	  XFsbl_Printf(DEBUG_INFO,"0x%08X: 0x",ireg);
+	  for(jreg = 0; jreg < 8; jreg++){
+	    XFsbl_Printf(DEBUG_INFO,"%08X ",XFsbl_In32(ireg));
+	    ireg+=4;
+	  }
+	  XFsbl_Printf(DEBUG_INFO,"\n\r");
+	}
+
+
 END:
 	return Status;
 }
